Sources/ComposableArchitecture/SharedState/PersistenceKey/AppStorageKey.swift

```diff
@@ -364,13 +364,36 @@ private protocol Lookup<Value> {
 }
 
 private struct CastableLookup<Value>: Lookup {
+  private let getValue: (_ key: String, _ store: UserDefaults) -> Value?
+  private let setValue: (Value?, _ key: String, _ store: UserDefaults) -> Void
```

Comment:
Instead of introducing these escaping closures, is it possible to add a new `Lookup` conformance just for `URL`?

Reply:
sure, see updated code